core/unit_test/TestComplex.hpp

```diff
@@ -532,4 +531,68 @@ TEST(TEST_CATEGORY, complex_operations_arithmetic_types_overloads) {
                               Kokkos::complex<long double>>::value));
 }
 
+TEST(TEST_CATEGORY, complex_structured_bindings) {
+  using Z = Kokkos::complex<double>;
+
+  // tuple_size
+  static_assert(std::is_same_v<std::tuple_size<Z>::type,
+                               std::integral_constant<size_t, 2>>);
+
+  // tuple_element
+  static_assert(std::is_same_v<std::tuple_element_t<0, Z>, Z::value_type>);
+  static_assert(std::is_same_v<std::tuple_element_t<1, Z>, Z::value_type>);
+
+  Z z(2., 3.);
+
+  // get lvalue
+  Z &l = z;
+  static_assert(std::is_same_v<decltype(Kokkos::get<0>(l)), Z::value_type &>);
+  static_assert(std::is_same_v<decltype(Kokkos::get<1>(l)), Z::value_type &>);
+  auto &[lr, li] = l;
+  ASSERT_EQ(lr, l.real());
+  ASSERT_EQ(li, l.imag());
```

Comment:
Does this also work in device code?

Reply:
Why wouldn't it?

Reply:
It seems to work fine, see https://godbolt.org/z/TGEvzYx5e (and our `Array` implementation). I'd still prefer if we also test in device code.

Reply:
Done.